src/main/java/uk/ac/cam/cl/dtg/isaac/dos/IsaacQuiz.java

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright 2021 Raspberry Pi Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ * 		http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.ac.cam.cl.dtg.isaac.dos;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import uk.ac.cam.cl.dtg.isaac.dto.IsaacQuizDTO;
+import uk.ac.cam.cl.dtg.segue.dos.content.ContentBase;
+import uk.ac.cam.cl.dtg.segue.dos.content.DTOMapping;
+import uk.ac.cam.cl.dtg.segue.dos.content.JsonContentType;
+import uk.ac.cam.cl.dtg.segue.dos.content.SeguePage;
+
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * DO for isaac quiz.
+ *
+ */
+@DTOMapping(IsaacQuizDTO.class)
+@JsonContentType("isaacQuiz")
+public class IsaacQuiz extends SeguePage {
+
+	private boolean visibleToStudents;
+
+	@JsonCreator
+	public IsaacQuiz(
+			@JsonProperty("id") String id, @JsonProperty("title") String title,
+			@JsonProperty("subtitle") String subtitle,
+			@JsonProperty("type") String type,
+			@JsonProperty("author") String author,
+			@JsonProperty("encoding") String encoding,
+			@JsonProperty("canonicalSourceFile") String canonicalSourceFile,
+			@JsonProperty("layout") String layout,
+			@JsonProperty("children") List<ContentBase> children,
+			@JsonProperty("value") String value,
+			@JsonProperty("attribution") String attribution,
+			@JsonProperty("relatedContent") List<String> relatedContent,
+			@JsonProperty("version") boolean published,
+			@JsonProperty("tags") Set<String> tags,
+			@JsonProperty("level") Integer level,
+			@JsonProperty("visibleToStudents") boolean visibleToStudents) {
+		super(id, title, subtitle, type, author, encoding,
+				canonicalSourceFile, layout, children, value, attribution,
+				relatedContent, published, tags, level);
+
+		this.visibleToStudents = visibleToStudents;
+	}
+
+	/**
+	 * Default constructor required for Jackson.
+	 */
+	public IsaacQuiz() {
+
+	}
+
+	public boolean getVisibleToStudents() {
+		return visibleToStudents;
+	}
+
+	public void setVisibleToStudents(boolean visibleToStudents) {
+		this.visibleToStudents = visibleToStudents;
+	}
+
+	@JsonIgnore
+	public List<IsaacQuizSection> getSections() {
```

Comment:
Maybe @mlt47 has opinions on this. As far as I remember, most of our DOs don't have logic in them, and this sort of stuff resides in the Manager layer, c.f. `GameManager::getAllMarkableQuestionPartsDFSOrder`?

Reply:
I don't even use this code yet, so maybe I should delete it?

I guess I saw this more as the contract of the DO (that it only contains IsaacQuizSection objects) and isn't actually intended to do any logic (the code should be a no-op).

Is there a way to distinguish production from staging/dev? In prod, it could do what it does above (filter out non-IsaacQuizSection); in other environments it could be an assertion that there are no non-IsaacQuizSection and throw a runtime exception if there are.

Also, it should probably cache its response so it appears more referentially transparent.

Another option is to override and narrow `getChildren` and override `setChildren` with a guard, but `setChildren` is final so I can't without changing `Content`.

The final alternative is to make this not inherit from SeguePage, which maybe makes sense as it isn't exactly a page and is never presented as one?

Reply:
Thanks. But yeah, for consistency, I'll remove it and do it in the manager.